### ISSUE TITLE
mz_dataflow_operator_reachability column type definitions are wrong

### ISSUE THREAD

**Issue by teskje** [MaterializeInc/materialize#15886]
### What version of Materialize are you using?

main

### How did you install Materialize?

Built from source

### What is the issue?

Reported [in Slack](https://materializeinc.slack.com/archives/C02PPB50ZHS/p1667512212357229):

```
materialize=> select * from mz_internal.mz_dataflow_operator_reachability;
ERROR:  internal error: column 0 is not of expected type ColumnType { scalar_type: List { element_type: Int64, custom_id: None }, nullable: false }: List([UInt64(7), UInt64(2)])
```

**Comment by frankmcsherry** [comment]
There are a few that are wrong. I started https://github.com/MaterializeInc/materialize/pull/15879 but it fails the ... upgrade tests. :D

**Comment by frankmcsherry** [comment]
In addition, the fourth column is wrong in this relation as well (the wrong type of integer again). I haven't pushed that to the PR yet.

**Comment by frankmcsherry** [comment]
Fixed by #15879
